### PR TITLE
A new parameter "Close path" has been added to the "Point to path" algorithm.

### PR DESCRIPTION
**Pull request by MrChebur**
New feature implemented as described here:  https://github.com/qgis/QGIS/issues/34527
Now this algorithm can check and close the line if necessary.
If this pull request will be accepted it is nescessary to add this text to the [docs](https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/vectorcreation.html#qgispointstopath):
> Close path	CLOSE_PATH	[boolean]	Default: False	If checked, the first and last point of the line will be connected

How it works:

How it works

### CODE REVIEW

**Comment by root676** [comment]
Good idea! But shouldn't the parameter reflect the algorithm name more: eg. `close path`

**Comment by MrChebur** [comment]
@root676 I am not a native English speaker. To me, "Point to path" sounds strange and I prefer "Point to line". If necessary, feel free to rename it.

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 1]
```diff
@@ -173,6 +169,10 @@ def processAlgorithm(self, parameters, context, feedback):
             f.setAttributes(attributes)
             line = [node[1] for node in vertices]
 
+            if close_line is True:
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:172]:
```suggestion
            if close_line:
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
 # -*- coding: utf-8 -*-
-
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:2]: Can you re-insert the removed spacing?

Some of the removed empty lines are required

**Review by m-kuhn — COMMENTED** [review]
Nice improvement!

Regarding the wording, I would stick to the existing taxonomy for consistence and use "Close path".

File: python/plugins/processing/algs/qgis/PointsToPaths.py (hunk 2 above)
Comment (MrChebur) [reply python/plugins/processing/algs/qgis/PointsToPaths.py:2]: done

**Review by m-kuhn — APPROVED** [review]
Thank you very much

File: python/plugins/processing/algs/qgis/PointsToPaths.py (hunk 2 above)
Comment (m-kuhn) [reply python/plugins/processing/algs/qgis/PointsToPaths.py:2]: I think this specific line is still missing?

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 3]
```diff
@@ -48,18 +48,19 @@
...
 class PointsToPaths(QgisAlgorithm):
     INPUT = 'INPUT'
+    CLOSE_PATH = 'CLOSE_PATH'
     GROUP_FIELD = 'GROUP_FIELD'
     ORDER_FIELD = 'ORDER_FIELD'
     DATE_FORMAT = 'DATE_FORMAT'
     OUTPUT = 'OUTPUT'
     OUTPUT_TEXT_DIR = 'OUTPUT_TEXT_DIR'
-
+
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:57]:
```suggestion

```

I think these spaces will need to be removed as well to make the linter checks happy.

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 4]
```diff
@@ -163,7 +166,7 @@ def processAlgorithm(self, parameters, context, feedback):
         for group, vertices in points.items():
             if feedback.isCanceled():
                 break
-
+
```
Comment (MrChebur) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:169]: del spaces (not line)

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 5]
```diff
@@ -133,10 +136,10 @@ def processAlgorithm(self, parameters, context, feedback):
         for current, f in enumerate(features):
             if feedback.isCanceled():
                 break
-
+                
             if not f.hasGeometry():
                 continue
-
+
```
Comment (MrChebur) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:142]: del spaces (not line)

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 6]
```diff
@@ -133,10 +136,10 @@ def processAlgorithm(self, parameters, context, feedback):
         for current, f in enumerate(features):
             if feedback.isCanceled():
                 break
-
+
```
Comment (MrChebur) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:139]: del spaces (not line)

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 7]
```diff
@@ -83,7 +85,7 @@ def initAlgorithm(self, config=None):
 
     def name(self):
         return 'pointstopath'
-
+
```
Comment (MrChebur) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:88]: del spaces (not line)

File: python/plugins/processing/algs/qgis/PointsToPaths.py (hunk 6 above)
Comment (m-kuhn) [reply python/plugins/processing/algs/qgis/PointsToPaths.py:139]:
```suggestion

```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 8]
```diff
@@ -48,18 +48,18 @@
...
     OUTPUT = 'OUTPUT'
     OUTPUT_TEXT_DIR = 'OUTPUT_TEXT_DIR'
-
     def group(self):
         return self.tr('Vector creation')
 
     def groupId(self):
         return 'vectorcreation'
-
+
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:62]:
```suggestion

```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 9]
```diff
@@ -133,7 +135,6 @@ def processAlgorithm(self, parameters, context, feedback):
         for current, f in enumerate(features):
             if feedback.isCanceled():
                 break
-
             if not f.hasGeometry():
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:138]:
```suggestion

            if not f.hasGeometry():
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 10]
```diff
@@ -48,18 +48,17 @@
...
 class PointsToPaths(QgisAlgorithm):
     INPUT = 'INPUT'
+    CLOSE_PATH = 'CLOSE_PATH'
     GROUP_FIELD = 'GROUP_FIELD'
     ORDER_FIELD = 'ORDER_FIELD'
     DATE_FORMAT = 'DATE_FORMAT'
     OUTPUT = 'OUTPUT'
     OUTPUT_TEXT_DIR = 'OUTPUT_TEXT_DIR'
-
     def group(self):
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:57]:
```suggestion

    def group(self):
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 11]
```diff
@@ -48,18 +48,17 @@
...
         return self.tr('Vector creation')
 
     def groupId(self):
         return 'vectorcreation'
-
     def __init__(self):
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:62]:
```suggestion

    def __init__(self):
```

File: python/plugins/processing/algs/qgis/PointsToPaths.py [hunk 12]
```diff
@@ -1,5 +1,4 @@
 # -*- coding: utf-8 -*-
-
 """
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointsToPaths.py:2]:
```suggestion

"""
```

**Comment by m-kuhn** [comment]
@MrChebur  the diff is clean now. No further patching :-)

**Comment by MrChebur** [comment]
@ m-kuhn Sorry, I'm quite new here. I don't really understand how to use github correctly. =)

**Comment by m-kuhn** [comment]
No worries. Happy to help. The contribution is a nice one.
If you intend to make more contributions, it's worth doing a bit of local setup to make your life easier. https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md

**Comment by alexbruy** [comment]
@MrChebur may I ask you to add tests too?

**Comment by MrChebur** [comment]
@alexbruy Should I create a new pull request for tests?

**Comment by alexbruy** [comment]
@MrChebur just add them to this PR. Thanks!

**Comment by MrChebur** [comment]
@alexbruy Done.

**Pull request merged**